src/main/java/com/solace/samples/java/patterns/GuaranteedBlockingPublisher.java

```diff
@@ -0,0 +1,152 @@
+
+    /** Main method. */
+    public static void main(String... args) throws IOException {
+        if (args.length < 3) {  // Check command line arguments
+            System.out.printf("Usage: %s <host:port> <message-vpn> <client-username> [password]%n%n", SAMPLE_NAME);
+            System.exit(-1);
+        }
+        System.out.println(API + " " + SAMPLE_NAME + " initializing...");
+
+        final Properties properties = new Properties();
+        properties.setProperty(TransportLayerProperties.HOST, args[0]);          // host:port
+        properties.setProperty(ServiceProperties.VPN_NAME,  args[1]);     // message-vpn
+        properties.setProperty(AuthenticationProperties.SCHEME_BASIC_USER_NAME, args[2]);      // client-username
+        if (args.length > 3) {
+            properties.setProperty(AuthenticationProperties.SCHEME_BASIC_PASSWORD, args[3]);  // client-password
+        }
+        //properties.setProperty(JCSMPProperties.GENERATE_SEQUENCE_NUMBERS, true);  // not required, but interesting
+        properties.setProperty(TransportLayerProperties.RECONNECTION_ATTEMPTS, "20");  // recommended settings
+        properties.setProperty(TransportLayerProperties.CONNECTION_RETRIES_PER_HOST, "5");
+        // https://docs.solace.com/Solace-PubSub-Messaging-APIs/API-Developer-Guide/Configuring-Connection-T.htm
+        // ready to connect now
+        final MessagingService messagingService = MessagingService.builder(ConfigurationProfile.V1)
+                .fromProperties(properties)
+                .build();
+        messagingService.connect();  // blocking connect
+        messagingService.addServiceInterruptionListener(serviceEvent -> {
+            logger.info("### SERVICE INTERRUPTION: "+serviceEvent.getCause());
+            //isShutdown = true;
+        });
+        messagingService.addReconnectionAttemptListener(serviceEvent -> {
+            logger.info("### RECONNECTING ATTEMPT: "+serviceEvent);
+        });
+        messagingService.addReconnectionListener(serviceEvent -> {
+            logger.info("### RECONNECTED: "+serviceEvent);
+        });
+        
+        // build the publisher object, starts its own thread
+        final PersistentMessagePublisher publisher = messagingService.createPersistentMessagePublisherBuilder()
+                .withDeliveryAckWindowSize(30)
+                .build();
+        publisher.start();
+        
+        ScheduledExecutorService statsPrintingThread = Executors.newSingleThreadScheduledExecutor();
+        statsPrintingThread.scheduleAtFixedRate(() -> {
+            System.out.printf("%s %s Published msgs/s: %,d%n",API,SAMPLE_NAME,msgSentCounter);  // simple way of calculating message rates
+            msgSentCounter = 0;
+        }, 1, 1, TimeUnit.SECONDS);
+        
+        System.out.println(API + " " + SAMPLE_NAME + " connected, and running. Press [ENTER] to quit.");
+        System.out.println("Publishing to topic '"+ TOPIC_PREFIX + API.toLowerCase() + 
```

Comment:
I had an internal debate when building the JCSMP samples a while back.  My rationale was that all of the API/messaging code/alerts/events resulted in logger output, but the sample program's status updates and user information messages would just be emitted to System.out since it's meant to be a console application.  Although I missed one or two places.

Reply:
leave it in with a comment to remove when that bug is fixed. Also please post a link to it on the bug in JIRA so it can be fixed. Just say "Please make a PR removing this 1 line before closing"

Reply:
Okay I'm fine with that. just wanted to make sure it wasn't accidental. Obviously go ahead and fix the couple of places you missed.